pkg/agent/apiserver/handlers/bgproute/handler.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bgproute
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"reflect"
+
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/apis"
+	"antrea.io/antrea/pkg/agent/controller/bgp"
+	"antrea.io/antrea/pkg/querier"
+)
+
+// HandleFunc returns the function which can handle queries issued by the bgproutes command.
+func HandleFunc(bq querier.AgentBGPPolicyInfoQuerier) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if bq == nil || reflect.ValueOf(bq).IsNil() {
+			// The error message must match the "FOO is not enabled" pattern to pass antctl e2e tests.
+			http.Error(w, "bgp is not enabled", http.StatusServiceUnavailable)
+			return
+		}
+
+		var ipv4Routes, ipv6Routes = false, false
+		rawQuery := r.URL.RawQuery
+		switch rawQuery {
+		case "ipv4-only=":
+			ipv4Routes = true
+		case "ipv6-only=":
+			ipv6Routes = true
+		case "":
+			ipv4Routes = true
+			ipv6Routes = true
+		default:
+			http.Error(w, "invalid query", http.StatusBadRequest)
+			return
+		}
```

Comment:
let's use this variation:
```go
		values := r.URL.Query()
		var ipv4Only, ipv6Only bool
		if values.Has("ipv4-only") {
				if values.Get("ipv4-only") != "" {
						// bad request
				}
				ipv4Only = true
		}
		if values.Has("ipv6-only") {
				if values.Get("ipv6-only") != "" {
						// bad request
				}
				ipv6Only = true
		}
		if ipv4Only && ipv6Only {
				// bad request
		}
		// ...
		bgpRoutes, err := bq.GetBGPRoutes(r.Context(), !ipv6Only, !ipv4Only)
```

I feel like the current validation is a bit too restrictive / low-level. We should accept both `ipv4-only=` and `ipv4-only` for example.